app/src/main/java/org/mozilla/focus/searchsuggestions/ui/SearchSuggestionsFragment.kt

```diff
@@ -0,0 +1,237 @@
+        }
+
+        dismiss_no_suggestions_message.setOnClickListener {
+            searchSuggestionsViewModel.dismissNoSuggestionsMessage()
+        }
+    }
+
+    private fun buildEnableSearchSuggestionsSubtitle(): SpannableString {
+        val subtitle = resources.getString(R.string.enable_search_suggestion_subtitle)
+        val appName = resources.getString(R.string.app_name)
+        val learnMore = resources.getString(R.string.enable_search_suggestion_subtitle_learnmore)
+
+        val spannable = SpannableString(String.format(subtitle, appName, learnMore))
+        val startIndex = spannable.indexOf(learnMore)
+        val endIndex = startIndex + learnMore.length
+
+        val learnMoreSpan = object : ClickableSpan() {
+            override fun onClick(textView: View) {
+                val url = "https://mozilla.org"
+                SessionManager.getInstance().createSession(Source.MENU, url)
+            }
+
+            override fun updateDrawState(ds: TextPaint?) {
+                ds?.isUnderlineText = false
+            }
+        }
+
+        spannable.setSpan(learnMoreSpan, startIndex, endIndex, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)
+        val color = ForegroundColorSpan(
+                ContextCompat.getColor(context!!, R.color.searchSuggestionPromptButtonTextColor))
+        spannable.setSpan(color, startIndex, endIndex, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)
+
+        return spannable
+    }
+
+    companion object {
+        fun create() = SearchSuggestionsFragment()
+    }
+
+    inner class SuggestionsAdapter(
+        private val clickListener: (String) -> Unit
+    ) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+        inner class DiffCallback(
+            private val oldSuggestions: List<SpannableStringBuilder>,
+            private val newSuggestions: List<SpannableStringBuilder>
+        ) : DiffUtil.Callback() {
+            override fun getOldListSize(): Int = oldSuggestions.size
+            override fun getNewListSize(): Int = newSuggestions.size
+            override fun areItemsTheSame(p0: Int, p1: Int): Boolean = true
+            override fun areContentsTheSame(p0: Int, p1: Int): Boolean = oldSuggestions[p0] == newSuggestions[p0]
+            override fun getChangePayload(oldItemPosition: Int, newItemPosition: Int): Any? = newSuggestions[newItemPosition]
+        }
+
+        private var suggestions: List<SpannableStringBuilder> = listOf()
+
+        fun refresh(suggestions: List<SpannableStringBuilder>) {
+            launch(CommonPool) {
+                val result = DiffUtil.calculateDiff(DiffCallback(this@SuggestionsAdapter.suggestions, suggestions))
+
+                launch(UI) {
+                    result.dispatchUpdatesTo(this@SuggestionsAdapter)
+                    this@SuggestionsAdapter.suggestions = suggestions
+                }
+            }
+        }
+
+        override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int, payloads: MutableList<Any>) {
```

Comment:
\[detekt\] INFO: \[style/ReturnCount\] Restrict the number of return statements in methods\.